File: 백준/Silver/2606. 바이러스/바이러스.py
# 2606
# 바이러스

import sys
from collections import deque

input = sys.stdin.readline

N = int(input())
V = int(input())

graph = [[] for _ in range(N+1)]

for i in range(V):
    a, b = map(int, input().split(" "))
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (N+1)
cnt = -1

def dfs(v):
    visited[v] = True
    global cnt
    cnt += 1

    for i in graph[v]:
        if not visited[i]:
            dfs(i)

dfs(1)
print(cnt)

File: 백준/Silver/2606. 바이러스/바이러스.java
import java.util.*;
import java.io.*;

public class Main {
    static boolean[][] graph;
    static boolean[] visited;
    static int res = 0;
    static int N, M;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        M = Integer.parseInt(br.readLine());
        graph = new boolean[N+1][N+1];
        visited = new boolean[N+1];

        int s, e = 0;
        String[] inputs;
        for(int i = 0; i < M; i++) {
            inputs = br.readLine().split(" ");
            s = Integer.parseInt(inputs[0]);
            e = Integer.parseInt(inputs[1]);

            graph[s][e] = graph[e][s] = true;
        }
        dfs(1);
        System.out.println(res);
    }

    public static void dfs(int v) {
        visited[v] = true;
        for (int i = 1; i <= N; i++) {
            if(graph[v][i] && !visited[i]) {
                res++;
                dfs(i);
            }
        }
    }
}